rules/manager.go

```diff
@@ -743,6 +743,33 @@ func (g *Group) RestoreForState(ts time.Time) {
 
 }
 
+// Equals return if two groups are the same
+func (g *Group) Equals(ng *Group) bool {
```

Comment:
This is going to be fragile when we add new fields, I'd suggest marshalling it and comparing that.

Reply:
Marshalling would require public fields.

Reply:
there are two options to avoid define extra Equals method:
1. expose public fields and allow marshalling
2. enhance Hash function to include fields.

for #2, it's not a good choice, because the Hash function is used to calculate the evaluation offset.